cli/internal/plugins/plugins.go

```diff
@@ -128,6 +128,33 @@ func (p *PluginManager) DownloadSource(ctx context.Context, spec *specs.Source)
 	}
 }
 
+func binarySuffix() string {
+	if runtime.GOOS == "windows" {
+		return ".exe"
+	}
+	return ""
+}
+
+func (p *PluginManager) getPaths(org string, repo string, registry string, version string) (destDir string, destPath string, zipPath string, binaryPathInZip string, downloadURL string) {
+	pluginName := fmt.Sprintf("cq-source-%s_%s_%s", repo, runtime.GOOS, runtime.GOARCH)
+	destDir = filepath.Join(p.directory, "plugins", registry, org, repo, version)
+	binarySuffix := binarySuffix()
+	destPath = filepath.Join(destDir, pluginName+binarySuffix)
+	zipPath = filepath.Join(destDir, pluginName+".zip")
+	binaryPathInZip = "plugins/source/" + repo + binarySuffix
+	downloadURL = fmt.Sprintf("https://github.com/cloudquery/cloudquery/releases/download/plugins/source/%s/%s/%s_%s_%s.zip", repo, version, repo, runtime.GOOS, runtime.GOARCH)
+	// we use convention over configuration and we use github as our registry. Similar to how terraform and homebrew work.
+	// For example:
+	// https://github.com/cloudquery/cloudquery/releases/download/plugins-source-test-v1.1.0/test_darwin_amd64.zip
+	if org != "cloudquery" {
+		// https://github.com/yevgenypats/cq-source-test/releases/download/v1.0.0/cq-source-test_linux_amd64.zip
+		downloadURL = fmt.Sprintf("https://github.com/%s/cq-source-%s/releases/download/%s/cq-source-%s_%s_%s.zip", org, repo, version, repo, runtime.GOOS, runtime.GOARCH)
+		binaryPathInZip = "cq-source-" + repo + binarySuffix
+	}
+
+	return destDir, destPath, zipPath, binaryPathInZip, downloadURL
```

Comment:
This might be a bit too much I know, but couldn't think of a nice way at the moment

Reply:
Maybe we can turn the return type into a struct? We're less likely to make mistakes related to the ordering of return values down the line then

Reply:
Makes sense, done in https://github.com/cloudquery/cloudquery/pull/1894/commits/4dda652c38d0555f620ad59b7539a32e310bd91e